server/src/main/java/org/opensearch/cluster/routing/allocation/ExistingShardsAllocator.java

```diff
@@ -60,6 +61,31 @@ public interface ExistingShardsAllocator {
         Setting.Property.PrivateIndex
     );
 
+    /**
+     * Boolean setting to enable/disable batch allocation of unassigned shards already existing on disk.
+     * This will allow sending all Unassigned Shards to the ExistingShard Allocator to  make decision to allocate
+     * in one or more go.
+     *
+     * Enable this setting if your ExistingShardAllocator is implementing the
+     * {@link ExistingShardsAllocator#allocateAllUnassignedShards(RoutingAllocation, boolean)} method.
+     * The default implementation of this method is not optimized and assigns shards one by one.
+     *
+     * If enable to true then it expects all indices of the shard to use same {@link ExistingShardsAllocator}, otherwise
+     * Allocation Service will fallback to default implementation i.e. {@link ExistingShardsAllocator#allocateUnassigned(ShardRouting, RoutingAllocation, UnassignedAllocationHandler)}
+     *
+     * If no plugin overrides {@link ExistingShardsAllocator} then default implementation will be use for it , i.e,
+     * {@link ShardsBatchGatewayAllocator}. Right now even if plugin implements it, AllocationService will run the
+     * default implementation to enable Batch mode of assignment
+     *
+     * TODO: Currently its implementation is WIP for GatewayAllocator so setting enabling wont have any effect
+     * https://github.com/opensearch-project/OpenSearch/issues/5098
```

Comment:
Right now and the below part is un-necessary

Reply:
you can say this setting is experimental at this point.

Reply:
Okay , will update